hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mob/mapreduce/MobRefReporter.java

```diff
@@ -0,0 +1,501 @@
+          }
+        } catch (IOException e) {
+          LOG.warn("Found file '{}' in archive area, but got an error while checking "
+              + "on back references.", file, e);
+          context.write(ARCHIVE_ERROR_IO, encodeRows(context, key, rows));
+        }
+
+      } else {
+        // check for an hlink in the active mob area (from a snapshot of a different table)
+        try {
+          /**
+           * we are doing this ourselves instead of using FSUtils.getReferenceFilePaths because
+           * we know the mob region never splits, so we can only have HFileLink references
+           * and looking for just them is cheaper then listing everything.
+           *
+           * This glob should match the naming convention for HFileLinks to our referenced hfile.
+           * As simplified explanation those file names look like "table=region-hfile". For details
+           * see the {@link HFileLink#createHFileLinkName HFileLink implementation}.
+           */
+          FileStatus[] hlinks = mob.getFileSystem(conf).globStatus(new Path(mob + "/*=*-" + file));
+          if (hlinks != null && hlinks.length != 0) {
+            if (hlinks.length != 1) {
+              LOG.warn("Found file '{}' as hfilelinks in the mob area, but there are more than " +
+                  "one: {}", file, Arrays.deepToString(hlinks));
+            }
+            HFileLink found = null;
+            for (FileStatus hlink : hlinks) {
+              HFileLink tmp = HFileLink.buildFromHFileLinkPattern(conf, hlink.getPath());
+              if (tmp.exists(archive.getFileSystem(conf))) {
+                found = tmp;
+                break;
+              } else {
+                LOG.debug("Target file does not exist for ref {}", tmp);
+              }
+            }
+            if (found != null) {
+              LOG.debug("Found file '{}' as a ref in the mob area: {}", file, found);
+              context.write(HLINK_CLONE, key);
+            } else {
+              LOG.warn("Found file '{}' as ref(s) in the mob area but they do not point to an hfile"
+                  + " that exists.", file);
+              context.write(HLINK_ERROR_DANGLING, encodeRows(context, key, rows));
+            }
+          } else {
+            LOG.error("Could not find referenced file '{}'. See the docs on this tool.", file);
+            LOG.debug("Note that we don't have the server-side tag from the mob cells that says "
+                + "what table the reference is originally from. So if the HFileLink in this table "
+                + "is missing but the referenced file is still in the table from that tag, then "
+                + "lookups of these impacted rows would still work.");
+            context.write(MISSING, encodeRows(context, key, rows));
+          }
+        } catch (IOException e) {
+          LOG.error("exception while checking mob area of our table for hfilelinks the point to {}",
```

Comment:
`exception[...]the point to` ->  `Exception while checking mob area of our table for HFileLinks that point to {}`?